### PR TITLE
Fix Servicehosting/Service.php

### LINKED ISSUE #771: Servicehosting/Service.php
ERROR
Argument 1 passed to Box\Mod\Servicehosting\Service::toHostingServerApiArray() must be an instance of Model_ServiceHostingServer, null given, called in /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-modules/Servicehosting/Service.php on line 449

Look for detailed error explanation

Set BB_DEBUG to FALSE, to hide the message below
Class: "TypeError"

File: "/var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-modules/Servicehosting/Service.php"

Line: "465"

Trace:
#0 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-modules/Servicehosting/Service.php(449): Box\Mod\Servicehosting\Service->toHostingServerApiArray()
#1 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-modules/Order/Service.php(1280): Box\Mod\Servicehosting\Service->toApiArray()
#2 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-modules/Order/Api/Client.php(86): Box\Mod\Order\Service->getOrderServiceData()
#3 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-library/Api/Handler.php(112): Box\Mod\Order\Api\Client->service()
#4 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/Extension/CoreExtension.php(1507): Api_Handler->__call()
#5 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-data/cache/e8/e8f55b3e35b21985b2b34a7d9562ca67d430f0b9ddf14cce2559a22b06f38a32.php(611): twig_get_attribute()
#6 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/Template.php(182): __TwigTemplate_0a901cef81e48aadcc8c536753a4abfc16a734da9bad4355d78ec035e95151b4->block_content()
#7 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-data/cache/f0/f0a3bc80c3e2cc260d8481167019bb61a31ce30b4fadbd168fc223bb2ad5e466.php(237): Twig\Template->displayBlock()
#8 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/Template.php(405): __TwigTemplate_4737e32b10cd3c7831d97dc9263374615c6d00c72dcd01a621f184f7937ce147->doDisplay()
#9 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#10 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-data/cache/e8/e8f55b3e35b21985b2b34a7d9562ca67d430f0b9ddf14cce2559a22b06f38a32.php(50): Twig\Template->display()
#11 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/Template.php(405): __TwigTemplate_0a901cef81e48aadcc8c536753a4abfc16a734da9bad4355d78ec035e95151b4->doDisplay()
#12 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#13 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/Template.php(390): Twig\Template->display()
#14 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render()
#15 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-library/Box/AppClient.php(79): Twig\TemplateWrapper->render()
#16 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-modules/Order/Controller/Client.php(86): Box_AppClient->render()
#17 [internal function]: Box\Mod\Order\Controller\Client->get_order()
#18 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-library/Box/App.php(199): ReflectionMethod->invokeArgs()
#19 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-library/Box/App.php(249): Box_App->executeShared()
#20 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/bb-library/Box/App.php(143): Box_App->processRequest()
#21 /var/www/vhosts/webbhostingservices.com/httpdocs/boxbilling/src/index.php(27): Box_App->run()
#22 {main}
Powered by BoxBilling

### PR DESCRIPTION
**Pull request by timothygwebb**
Close #771

### CODE REVIEW

File: src/.htaccess [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # If you are receiving "Internal Server Error" message, make sure mod_rewrite is enabled in apache
 
 Options -Indexes
-Options +FollowSymLinks
+# Options +FollowSymLinks
```
Comment (evrifaessa) [inline src/.htaccess:5]: And this is for..?

**Comment by timothygwebb** [comment]
It is not allowed on my system.

Please discuss the following and update appropriately.

If you use Options directive in .htaccess with parameter which has been
forbidden in main Apache config, server will return HTTP 500 error code.

Allowed .htaccess options are defined by directive AllowOverride in the
main Apache config file. To allow symlinks, this directive need to be set
to All or Options.

Besides allowing use of symlinks, this directive is also needed to enable
mod_rewrite in .htaccess context. But for this, also the more secure
SymLinksIfOwnerMatch option can be used.

File: src/.htaccess (hunk 1 above)
Comment (timothygwebb) [reply src/.htaccess]: Added SymLinksIfOwnerMatch

**Review by timothygwebb — COMMENTED** [review]
Added SymLinksIfOwnerMatch

**Pull request merged**